### PR TITLE
Generate Protobuf in sbt

### PR DESCRIPTION
**Pull request by ANeumann82**
PR to include protobuf generation in the SBT build process.

Upside:
- We don't need to do it manually anymore, and the Protos.java is not included in the version control anymore.

Downside:
- Adds some build dependencies
  - The actual SBT plugin: https://github.com/sbt/sbt-protobuf
  - ProtoC-Provider for Java: https://github.com/os72/protoc-jar

We do download and execute the remote protobuf binary, but... Well, we do that with dependencies all the time.

### CODE REVIEW

**Comment by ANeumann82** [comment]
Please let me know your thoughts about it. I think it's a nice to have convenience feature, but nothing I would insist on.

File: build.sbt [hunk 1]
```diff
@@ -68,6 +65,12 @@ lazy val integrationTestSettings = Seq(
   concurrentRestrictions in Test := Seq(Tags.limitAll(math.max(1, java.lang.Runtime.getRuntime.availableProcessors() / 2)))
 )
 
+val pbSettings = ProtobufPlugin.projectSettings ++ Seq(
```
Comment (jeschkies) [inline build.sbt]: Could you add some doc? It would make it a little easier to maintain.

File: src/main/protobuf/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Google Protocol Buffers
+
+Marathon is using Google [Protocol Buffers](https://developers.google.com/protocol-buffers) to marshal and unmarshal data from the persistent store.
+
+The Protos are now automatically updated in the sbt build process.
```
Comment (jeschkies) [inline src/main/protobuf/README.md]:
```suggestion
The Protos are automatically updated in the sbt build process.
```

File: build.sbt (hunk 1 above)
Comment (ANeumann82) [reply build.sbt]: Done

File: src/main/protobuf/README.md (hunk 2 above)
Comment (ANeumann82) [reply src/main/protobuf/README.md]: Done

**Pull request merged**